Misc/NEWS.d/next/Core_and_Builtins/2024-08-27-21-44-14.gh-issue-116017.ZY3yBY.rst

```diff
@@ -0,0 +1 @@
+Improved JIT memory consumption by invalidating cold executors
```

Comment:
Since this is user-facing, I'd reword it to avoid mentioning implementation details (nobody knows what an "executor" is) and focus on the user-facing impact. For example, I'd emphasize that the improvement happens because the memory for "cold" (or infrequently-executed) code will now be freed periodically (instead of living forever). This is pretty cool, so don't be afraid to write a few sentences if it would help explain how programs will benefit, etc. :)